src/System.Net.Http/src/System/Net/Http/Managed/ChunkedEncodingReadStream.cs

```diff
@@ -26,7 +26,16 @@ public ChunkedEncodingReadStream(HttpConnection connection)
                 Debug.Assert(_chunkBytesRemaining == 0);
 
                 // Start of chunk, read chunk size.
-                ulong chunkSize = ParseHexSize(await _connection.ReadNextLineAsync(cancellationToken).ConfigureAwait(false));
+                ArraySegment<byte> line;
+                while (!_connection.TryReadNextLine(out line))
```

Comment:
I know it'll mess up the pattern here, but I'm curious if you see any throughput benefits by changing the signature to return the `ArraySegment<byte>` with an `out bool`, or if you return a `(bool, ArraySegment<byte>)`.  I'm specifically wondering about the write barriers that may be incurred from writing to the out array (we can check the asm to verify there is one).